google/cloud/spanner_v1/merge_result_set.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2024 Google LLC All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from concurrent.futures import ThreadPoolExecutor
+from dataclasses import dataclass
+from queue import Queue
+from typing import Any, TYPE_CHECKING
+from threading import Lock, Semaphore
+
+if TYPE_CHECKING:
+    from google.cloud.spanner_v1.database import BatchSnapshot
+
+QUEUE_SIZE_PER_WORKER = 32
+MAX_PARALLELISM = 100
```

Comment:
My impression is that this is too high, especially when it is a constant. 100 open query streams that are being consumed in parallel (at least partly in parallel, depending on how often they are blocked on IO), each being able to push 32 rows into the queue, could still consume much memory if the rows are large. I would rather limit this to 16 for now, and do a follow-up where we make this a variable that the user can set.

Reply:
Changed it to 16